Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -363,14 +363,21 @@ function Publish-UpgradedServiceFabricApplication
         {
             Write-Host (Get-VstsLocString -Key SFSDK_UpgradeSuccess)
         }
-        elseif ($upgradeStatus.UpgradeState -eq "RollingBackCompleted")
+        else
         {
             $global:operationId = $SF_Operations.WaitApplicationUpgradeStatus
 
             Trace-ServiceFabricApplicationHealth -ApplicationName $ApplicationName
             Trace-ServiceFabricClusterHealth
 
-            Write-Error (Get-VstsLocString -Key SFSDK_UpgradeRolledBack)
+            if ($upgradeStatus.UpgradeState -eq "RollingBackCompleted")
+            {
+                Write-Error (Get-VstsLocString -Key SFSDK_UpgradeRolledBack)
+            }
+            else
```

Comment:
No need to write ELSE statement, because code will reach till this point only when it doesn't not enter in IF statement. And in case when it does enter in IF then task will get terminated after executing Write-Error statement.